### PR TITLE
Create editionable worldwide org

### PR DESCRIPTION
**Pull request by jkempster34**
https://trello.com/c/3WmL0sbs

### Add basic editionable worldwide organisation
The existing worldwide organisation modelling presents various issues.
Worldwide organisations surface the main body of their content through an
associated corporate information about us page. Issues include:
- Draft pages don't update correctly #8358
- We redirect from the base path of the corporate information page back to the
  worldwide organisation #8037

We want to add new "editionable" worldwide organisations in order to improve
this area of Whitehall. The intention being that these eventually replace the
existing worldwide organisation models.

This adds the basic routes, model and controller actions. The actions are
behind a feature flag.

No publishing API presenter is configured yet.

---

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/models/worldwide_organisation_edition.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class WorldwideOrganisationEdition < Edition
+  include Edition::Organisations
```
Comment (jkempster34) [inline app/models/worldwide_organisation_edition.rb:2]: This is interesting, I think all editions have related organisations which can be filtered on here 👇 We think this is desirable and we would probably migrate the existing ones all to be FCDO
image

**Review by jkempster34 — COMMENTED** [review]
Worth noting that this is meant to be a minimal first step so it currently doesn't publish anything (which should be okay because this is behind a feature flag)

The next step is to probably write a Publishing API Presenter for that purpose

**Review by mike29736 — APPROVED** [review]
Given that you've explained that this is just "the basics" and given that it's behind a feature flag, and with the caveat that I don't have the background knowledge to spot anything that's missing (!), nothing here stands out to me as unreasonable or broken :+1:

**Comment by mike29736** [comment]
> [...] or broken 👍

Other than that `erblint` CI step

File: app/helpers/admin/new_document_helper.rb [hunk 2]
```diff
@@ -10,11 +10,13 @@ module Admin::NewDocumentHelper
     CaseStudy,
     StatisticalDataSet,
     CallForEvidence,
+    WorldwideOrganisationEdition,
   ].freeze
 
   def new_document_type_list
     NEW_DOCUMENT_LIST
       .select { |edition_type| can?(:create, edition_type) }
+      .reject { |edition_type| edition_type == WorldwideOrganisationEdition && !Flipflop.editionable_worldwide_organisations? }
```
[the remark this answers is not in the document]
Comment (jkempster34) [reply app/helpers/admin/new_document_helper.rb:19]: I'll explore that yeah, looks like we can set rules per model type

**Comment by jkempster34** [comment]
> > [...] or broken 👍
> 
> Other than that `erblint` CI step

Didn't notice that sorry!

File: app/models/worldwide_organisation_edition.rb (hunk 1 above)
Comment (JonathanHallam) [reply app/models/worldwide_organisation_edition.rb:2]: There are a few of these that are owned by other organisations, I think the MOD have a couple so we'd have to ask around a bit before making this decision

File: app/models/worldwide_organisation_edition.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+  def display_type_key
+    "worldwide_organisation_edition"
```
Comment (JonathanHallam) [inline app/models/worldwide_organisation_edition.rb:5]: I was really hoping this was used to make the name in the new document section, would have been an easy way around the type display issue

File: app/controllers/admin/worldwide_organisation_editions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class Admin::WorldwideOrganisationEditionsController < Admin::EditionsController
```
[the remark this answers is not in the document]
Comment (JonathanHallam) [reply app/controllers/admin/worldwide_organisation_editions_controller.rb:1]: I think it's also extensible, so in the new document helper we can strip `editionable` from the start of a string if it exists, means we could extend this if we ever do it again, like I said, happy to be talked out of this as i realise we could just strip it from the end

**Comment by JonathanHallam** [comment]
One comment but otherwise looking great, thanks

File: app/models/worldwide_organisation_edition.rb (hunk 1 above)
Comment (jkempster34) [reply app/models/worldwide_organisation_edition.rb]: Ah, so these might replace the "sponsoring organisations"?

File: app/helpers/admin/new_document_helper.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkempster34) [reply app/helpers/admin/new_document_helper.rb]: Done! The base EditionsController is also configured to response with a forbidden status code if `can?` is false. Thanks!

File: app/controllers/admin/worldwide_organisation_editions_controller.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (jkempster34) [reply app/controllers/admin/worldwide_organisation_editions_controller.rb]: Done 👍

**Comment by jkempster34** [comment]
> Ooh, as a slight aside would you mind adding instructions in the PR description on how to set the feature flag up in a docker environment? I had to delete the feature flag changes to see it because I was too lazy to work it out

There is a doc here https://github.com/alphagov/whitehall/blob/main/docs/feature_flags.md

**Pull request merged**